src/crimson/osd/heartbeat.cc

```diff
@@ -199,12 +199,10 @@ seastar::future<> Heartbeat::remove_peer(osd_id_t peer)
   auto found = peers.find(peer);
   assert(found != peers.end());
   logger().info("remove_peer({})", peer);
-  return seastar::when_all_succeed(found->second.con_front->close(),
-                                   found->second.con_back->close()).then(
-    [this, peer] {
-      peers.erase(peer);
-      return seastar::now();
-    });
+  found->second.con_front->close();
+  found->second.con_back->close();
+  peers.erase(peer);
+  return seastar::now();
```

Comment:
`Heartbeat::remove_peer()` doesn't need to return `future<>`, And there are chained impacts to other implementations, see https://github.com/ceph/ceph/pull/33909/commits/1d34640a47eb46d7958b5960d8316a89681f8bef.